app/src/main/java/com/github/geohunt/app/ServiceLocator.kt

```diff
@@ -0,0 +1,18 @@
+package com.github.geohunt.app
+
+import com.github.geohunt.app.authentication.Authenticator
+import com.github.geohunt.app.authentication.FirebaseAuthenticator
+
+class ServiceLocator {
+    companion object {
+        private var authenticator: Authenticator = FirebaseAuthenticator()
+
+        fun getAuthenticator(): Authenticator {
+            return authenticator
+        }
+
+        fun setAuthenticator(authenticator: Authenticator) {
+            ServiceLocator.authenticator = authenticator
+        }
+    }
+}
```

Comment:
I believe having a single `ServiceLocator` for all classes will become a mess, wouldn't it better to have a singleton in each interface that appear here ?

For instance here we could have something like
```kotlin
// Authenticator (interface)
companion object {
    val authenticatorInstance = Singleton<Authenticator> {
        FirebaseAuthenticator()
    }
}
```

It allows also for declaring object with non-trivial constructor (by having a singleton of a factory instead). This `Singleton<(Args...) -> Object>` can then be mocked before the instantiation of the test to allow for mocked object (cf the database in my PR #36)